v2/pkg/engine/datasource/grpc_datasource/execution_plan_visitor.go

```diff
@@ -819,8 +899,16 @@ func (r *rpcPlanVisitor) parseGraphQLType(t *ast.Type) DataType {
 	}
 }
 
-func (r *rpcPlanVisitor) isNullableScalar(fdt int) bool {
-	return r.definition.TypeIsScalar(fdt, r.definition) && !r.definition.TypeIsNonNull(fdt)
+func (r *rpcPlanVisitor) typeIsNullableOrNestedList(typeRef int) bool {
+	if !r.definition.TypeIsNonNull(typeRef) && r.definition.TypeIsList(typeRef) {
+		return true
```

Comment:
Should this helper care only about nested lists? 
because for `[string]` you will return true here, but it won't be nested list

Reply:
No, both cases are relevant for me as nullable AND nested lists use the wrapper type.